### PR TITLE
added validation endpoint + cleanup of possible responses

### PR DESCRIPTION
**Pull request by kralle333**
closes dfds/cloudplatform/issues/2316

# Additional Review Notes
Also added an ids endpoint to make it easy from the outside to see valid object ids

### CODE REVIEW

File: src/SelfService/Infrastructure/Api/JsonSchema/SelfServiceJsonSchemaController.cs [hunk 1]
```diff
@@ -59,10 +59,9 @@ public async Task<IActionResult> GetSchema(string id, [FromQuery] int schemaVers
 
     [HttpPost("{id:required}")]
     [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest, "application/problem+json")]
     [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status401Unauthorized, "application/problem+json")]
```
Comment (andreasfrisch) [inline src/SelfService/Infrastructure/Api/JsonSchema/SelfServiceJsonSchemaController.cs:63]: Is this ever used?
Should it be?

File: src/SelfService/Infrastructure/Api/JsonSchema/SelfServiceJsonSchemaController.cs [hunk 2]
```diff
@@ -121,6 +120,44 @@ public async Task<IActionResult> AddSchema(string id, [FromBody] AddSelfServiceJ
                 new ProblemDetails { Title = "Uncaught Exception", Detail = $"AddSchema: {e.Message}." }
             );
         }
-#pragma warning restore CS0162 // Unreachable code detected
+    }
+
+    [HttpPost("validate")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status401Unauthorized, "application/problem+json")]
```
Comment (andreasfrisch) [inline src/SelfService/Infrastructure/Api/JsonSchema/SelfServiceJsonSchemaController.cs:127]: This is never actually returned, is it?

File: src/SelfService/Infrastructure/Api/JsonSchema/SelfServiceJsonSchemaController.cs [hunk 3]
```diff
@@ -121,6 +120,44 @@ public async Task<IActionResult> AddSchema(string id, [FromBody] AddSelfServiceJ
...
+        }
+        catch (Exception e)
+        {
+            return BadRequest(e);
+        }
+
+        return Ok();
+    }
+
+    [HttpGet("ids")]
```
Comment (andreasfrisch) [inline src/SelfService/Infrastructure/Api/JsonSchema/SelfServiceJsonSchemaController.cs:157]: when is this route supposed to be used?

**Review by andreasfrisch — APPROVED** [review]
Nothing blocking;
We don't check for authorization.
That is not dangerous per se in these cases, so it's not a critical issue.

Nice to have a good way to remotely validate schemas 💪

File: src/SelfService/Infrastructure/Api/JsonSchema/SelfServiceJsonSchemaController.cs (hunk 3 above)
Comment (kralle333) [reply src/SelfService/Infrastructure/Api/JsonSchema/SelfServiceJsonSchemaController.cs:157]: I think I will just remove it. Just found it useful, but should be easy enough to add again when we actually need it.

File: src/SelfService/Infrastructure/Api/JsonSchema/SelfServiceJsonSchemaController.cs (hunk 1 above)
Comment (kralle333) [reply src/SelfService/Infrastructure/Api/JsonSchema/SelfServiceJsonSchemaController.cs:63]: image
First check can return BadRequest

**Pull request merged**